File: Data Structures & Algorithms/binary-search/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let lastDirection = 0;
        let i = Math.floor(nums.length/2);
        while (i < nums.length && i > -1) {
            console.log(nums[i]);
            if (nums[i] == target) {
                return(i);
            }
            if (nums[i] > target) {
                if (lastDirection == 1) return -1;
                i--;
                lastDirection = -1;
            }
            else if (nums[i] < target) {
                if (lastDirection == -1) return -1;
                i++;
                lastDirection = 1;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-5.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        store = {}
        for i in range(len(nums)):
            print(store)
            if target - nums[i] in store:
                return [store[target-nums[i]],i]
            else:
                store[nums[i]] = i
        print(store)
        return False

                